### PR TITLE
Fix webp animated image frame over lapping

### PR DESCRIPTION
**Pull request by wsdwsd0829**
The webp interpreter is trying to draw both previous and current frame to context resulting:
Image_11
https://lh3.googleusercontent.com/z2B73tA2ofYnFho0sj2KEQpze1fYY7RFSh5Wa9uElu67vF0R6wB59S2Roe3mzqbca5VHA2EaTTmEjtIxyeN9=s208-rwa

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by wsdwsd0829** [comment]
This pr just to start the conversation. 
I feel like we do not need to do the extra work? 
https://github.com/pinterest/PINRemoteImage/blob/master/Source/Classes/AnimatedImages/PINWebPAnimatedImage.m#L142 but there may be other cases depends on that? 
So 
option1: I suggest remove the previousFrame parameter if it's safe to do so.  (preferred)
option2: add a bool to control it so we can disable it, where do u suggest to control it best? (maybe ASAnimatedImageProtocol in ASImageProtocols.h ?)

**Comment by garrettmoon** [comment]
Hmm, my understanding of the spec is you can only not draw the previous frame if the current frame is a key frame. And that is taken into account in the code currently. I wonder if there's some other issue with transparency or background color?

**Comment by wsdwsd0829** [comment]
I tested a couple of images and seems fine, do you have any examples for me to test against. 
the issue is we cannot draw previous and current frame together. (https://github.com/pinterest/PINRemoteImage/blob/master/Source/Classes/AnimatedImages/PINWebPAnimatedImage.m#L142)  which will result in the image in description.

**Comment by wsdwsd0829** [comment]
Earlier attempt do break some cases because image blending logic is much more complex than expected. 
frameRequiresBlendingWithPreviousFrame is borrowed from some other framework.
I tested about 20 images that either/both breaks with current lib or first attempt. 
After the fix all looks good, I will post the test links once get permission :)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by garrettmoon** [comment]
@wsdwsd0829 Lemme know when you get a chance to post the test images!

**Comment by wsdwsd0829** [comment]
Sent to your email.

File: Source/Classes/AnimatedImages/PINWebPAnimatedImage.m [hunk 1]
```diff
@@ -278,6 +385,38 @@ - (CGImageRef)imageAtIndex:(NSUInteger)index cacheProvider:(nullable id<PINCache
     return canvas;
 }
 
+- (BOOL)frameRequiresBlendingWithPreviousFrame:(NSUInteger)index
```
Comment (garrettmoon) [inline Source/Classes/AnimatedImages/PINWebPAnimatedImage.m]: This seems very similar to the isKeyFrame logic… can that just be updated instead?

**Comment by garrettmoon** [comment]
I'm going to close this for now, lemme know if I missed anything in my PR!

**Pull request closed without merging**